textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/DictionaryPredicate.java

```diff
@@ -0,0 +1,81 @@
+package edu.uci.ics.textdb.dataflow.common;
+
+import java.util.List;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IDictionary;
+import edu.uci.ics.textdb.api.common.IPredicate;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.common.constants.DataConstants;
+import edu.uci.ics.textdb.common.constants.DataConstants.SourceOperatorType;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.dataflow.source.ScanBasedSourceOperator;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
+import edu.uci.ics.textdb.storage.reader.DataReader;
+
+public class DictionaryPredicate implements IPredicate {
+
+    private IDictionary dictionary;
+    private Analyzer analyzer;
+    private List<Attribute> attributeList;
+    private IDataStore dataStore;
+    private SourceOperatorType srcOpType;
+    
+    /*
+    dictionary refers to list of phrases to search for.
+    For Ex. New York if searched in TextField, we would consider both tokens
+    New and York; if searched in String field we search for Exact string.
+     */
+
+    public DictionaryPredicate(IDictionary dictionary, Analyzer analyzer, List<Attribute> attributeList,
+            SourceOperatorType srcOpType, IDataStore dataStore) {
+
+        this.dictionary = dictionary;
+        this.analyzer = analyzer;
+        this.attributeList = attributeList;
+        this.srcOpType = srcOpType;
```

Comment:
@zuozhi this is part - 1 of the refactoring implementation. Those changes will come in the next PR. :)